external-resolver/src/test/scala/org/genivi/sota/resolver/test/random/Query.scala

```diff
@@ -33,6 +35,24 @@ object Query extends
     FilterRequestsHttp with
     ResolveRequestsHttp {
 
+  def semQuerys(qrs: List[Query])
+               (implicit ec: ExecutionContext): State[RawStore, List[Semantics]] = {
+
+    @tailrec def go(qrs0: List[Query], s0: RawStore, acc: List[Semantics]): (RawStore, List[Semantics]) =
+      qrs0 match {
+        case Nil           => (s0, acc.reverse)
+        case (qr :: qrs1) =>
+          val (s1, r) = semQuery(qr).run(s0).run
+          go(qrs1, s1, r :: acc)
+      }
+
+    State.get.flatMap { s0 =>
```

Comment:
If I remember correctly, this piece of code can't be rewritten using for comprehensions!